android/app/src/main/java/com/couchbase/todo/db/DocDumper.java

```diff
@@ -0,0 +1,26 @@
+package com.couchbase.todo.db;
+
+import android.os.AsyncTask;
+import android.util.Log;
+
+import androidx.annotation.WorkerThread;
+
+import com.couchbase.lite.Document;
+
+import java.util.List;
+
+
+abstract class DocDumper<T> extends AsyncTask<T, Void, Exception> {
+    public static final String DOC_LOGGER = "DOC DUMP";
+    public static final String NULL_WARNING_LOGGER = "NULL DOC";
+
+    @WorkerThread
+    protected final void dumpDocs(List<String> docIds) {
+        for (String id: docIds) {
+            Document doc = DAO.get().fetch(id);
+            if (doc == null) { Log.w(NULL_WARNING_LOGGER, "A document is null"); }
```

Comment:
Interesting.  The, sort of, "normal" use of the "tag" argument, to the Android logger, is to identify, roughly, the file (or, at least, the component) that is logging the message.  Here, you'd probably want to use DOC_LOGGER for all of these logs, since they are coming from the doc logger.

It is even traditional to define a constant called "TAG" in a component, so that log messages *always* read `Log.d(TAG, ...)`